### PR TITLE
sourmash 3.x protein ksizes must be divisible by 3

### PR DESCRIPTION
**Pull request by bluegenes**
When I wrote #575  / #576 , I thought that protein sigs were calculated at their exact ksize, except when translating (I forgot about this PR until recently). So #576 eliminated a check for the protein ksize being divisible by 3, allowing you to calculate signatures with k=19, which would evaluate to 19/3=6.33, giving a protein ksize of 6. 

While this is not terrible, I think disabling these non-divisible (to integers) ksizes is informative for what is happening under the hood, and is thus clearer.

This PR simply reverses what was changed in #576, and removes the unnecessary (and now failing) test. If we change protein ksize calculation in sourmash 4.x, we can add these right back in.

- [x] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
hi @bluegenes could you explain the rationale for this a little bit more in the main PR text?

**Comment by bluegenes** [comment]
@ctb done!

**Comment by ctb** [comment]
hi @bluegenes as [described](https://github.com/dib-lab/sourmash/pull/1013#issuecomment-643276301) I merged #1013 after switching ksizes over to 57. I then merged that master back into this PR, and wanted to double check with you that this all looks right.

I actually just double checked myself, and it looks like commit 3ec04b5b9 is now all that's left on this PR, which seems appropriate. Yay!

Instead of removing the test that now fails, `test_do_sourmash_compute_multik_protein_input_non_div3_ksize`, what do you think about changing it so that it works correctly? I think this would involve asserting that the result of `runscript` is failure (a nonzero exit status).

**Comment by bluegenes** [comment]
@ctb - done! test mirrors the "bad ksize" test for nucleotide input.

File: tests/test_sourmash_compute.py [hunk 1]
```diff
@@ -587,7 +587,8 @@ def test_do_sourmash_compute_multik_protein_input_non_div3_ksize():
                                            in_directory=location,
                                            fail_ok=True)
         outfile = os.path.join(location, 'short-protein.fa.sig')
-        assert os.path.exists(outfile)
+        assert not os.path.exists(outfile)
```
Comment (ctb) [inline tests/test_sourmash_compute.py:590]:
```suggestion
        assert status != 0
```

**Comment by ctb** [comment]
great thx! one suggested change.

**Comment by ctb** [comment]
Before merging, I need to think about whether this violates our command-line interface semantic versioning guarantee :)

**Comment by ctb** [comment]
I think this may be a specific example of an API-breaking change that is ready to merge but would break 3.x! How exciting :)

Briefly, this check was removed several versions ago (December 2019), which doesn't violate semantic versioning requirements, since making command line checks more liberal is fine. To add it back in could, in theory, break people's workflows. Ugh!

So I'll put this in cold storage for a little bit while we settle into our 4.0-targeted groove viz #1016.

**Comment by ctb** [comment]
(i.e. please don't merge :)

**Comment by ctb** [comment]
merged in #1277.

**Pull request closed without merging**